src/QuickMarcEditor/QuickMarcEditor.js

```diff
@@ -31,41 +43,51 @@ const QuickMarcEditor = ({ instance, onClose, handleSubmit, initialValues, submi
       pristine={pristine}
     />
   ), [onClose, handleSubmit, submitting, pristine]);
-
   const initialRecords = initialValues?.records;
 
   useEffect(() => {
     if (initialRecords) setRecords(initialRecords);
   }, [initialRecords]);
 
+  const addRecord = useCallback(({ values }) => {
+    setRecords(values.records);
+  }, [records.length]);
```

Comment:
why do you check `records.length`? it will just create new callback when length is changed, but it will update records each time values are changed :-1: 

again do you test code you implement?